src/Fixer/Casing/NativeConstTypeDeclarationCasingFixer.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+final class NativeConstTypeDeclarationCasingFixer extends AbstractFixer
+{
+    /*
+     * https://wiki.php.net/rfc/typed_class_constants
+     * Supported types
+     * Class constant type declarations support all type declarations supported by PHP,
+     * except `void`, `callable`, `never`.
+     *
+     * array
+     * bool
+     * callable
+     * float
+     * int
+     * iterable
+     * object
+     * mixed
+     * parent
+     * self
+     * string
+     * any class or interface name -> not native, so not applicable for this Fixer
+     * ?type -> not native, `?` has no casing, so not applicable for this Fixer
+     *
+     * Not in the list referenced but supported:
+     * null
+     */
+    private const SUPPORTED_HINTS = [
+        'array' => true,
+        'bool' => true,
+        'float' => true,
+        'int' => true,
+        'iterable' => true,
+        'mixed' => true,
+        'null' => true,
+        'object' => true,
+        'parent' => true,
+        'self' => true,
+        'string' => true,
+    ];
+
+    private const TYPE_SEPARATION_TYPES = [
+        CT::T_TYPE_ALTERNATION,
+        CT::T_TYPE_INTERSECTION,
+        CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN,
+        CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE,
+    ];
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Native type hints for constants should use the correct case.',
```

Comment:
why only for constants? why not for all type hints?
eg `private INT $status;` could also be fixed.